### PR TITLE
Show timestamp of when user last made progress in a script

### PR DESCRIPTION
**Pull request by davidsbailey**
## Background

Finishes https://codedotorg.atlassian.net/browse/LP-1323

## Description

Today, in the progress view, we fetch all progress for all levels in the selected script, for each student in the section. In this PR, we start also sending down a timestamp for each user, indicating the most recent timestamp for any progress made across the levels in the script. We do not send down the progress timestamp for each level, to keep the data size small. However, we might start sending down per-level timestamps in the future.

### Screenshots

Screen Shot 2020-03-23 at 5 09 38 PM

Screen Shot 2020-03-23 at 5 09 45 PM

Screen Shot 2020-03-23 at 5 10 14 PM

Screen Shot 2020-03-23 at 5 10 21 PM

## Testing story

Added new unit tests. Manually tested end-to-end.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by Erin007 — APPROVED** [review]
Thanks for getting this out so quickly!

File: apps/src/templates/sectionProgress/SectionProgress.jsx [hunk 1]
```diff
@@ -137,8 +146,32 @@ class SectionProgress extends Component {
...
+        </span>
+      </ReactTooltip>
+    ));
+
+    return lessonTooltips.concat(studentTooltips);
   }
 
+  tooltipTextForStudent = studentId => {
+    const timestamp = this.props.studentTimestamps[studentId];
+    return timestamp ? moment(timestamp).calendar() : i18n.none();
```
Comment (islemaster) [inline apps/src/templates/sectionProgress/SectionProgress.jsx:172]: TIL `moment.calendar()`

**Comment by Erin007** [comment]
I wonder about dates in the month/day/year format; is there a way to switch do that based on locale?

**Review by islemaster — APPROVED** [review]
This is awesome.

**Comment by davidsbailey** [comment]
@Erin007 yes! moment accepts a locale argument. I think this is a good idea to ship today to unblock US classrooms. then I'll ship locale settings in a follow-up PR.

File: apps/src/templates/sectionProgress/SectionProgress.jsx (hunk 1 above)
Comment (clareconstantine) [reply apps/src/templates/sectionProgress/SectionProgress.jsx:172]: +1 :)

**Review by clareconstantine — APPROVED** [review]

**Pull request merged**